### PR TITLE
add to collections on upload

### PR DESCRIPTION
**Pull request by tsop14**
#1590 Adds ability to add/remove images to/from collections on upload page.

Remembers open/closed nodes and position on collections tree. 

Can batch add collections on one image to all recent uploads. 
Can remove all collections by 'adding' no collections (as per labels). 

! If the same image is upload again and it was previously added to collections, it will only display the collections once it is added to a collection. So, the collection is present in the media-collection api but not on the image's collections data in the media api and it only fetches the collections list when collections are edited.

collections on uploads

### CODE REVIEW

**Comment by kenoir** [comment]
Nice.

File: kahuna/public/js/edits/image-editor.js [hunk 1]
```diff
@@ -5,13 +5,17 @@ import './service';
...
+import {rememberScrollTop} from '../directives/gr-remember-scroll-top';
 
 
 export var imageEditor = angular.module('kahuna.edits.imageEditor', [
     'kahuna.edits.service',
     'gr.image.service',
     'kahuna.edits.usageRightsEditor',
-    'gr.archiverStatus'
+    'gr.archiverStatus',
+    collectionsApi.name,
```
Comment (kenoir) [inline kahuna/public/js/edits/image-editor.js]: Why this way and not the name as a string?

**Comment by paperboyo** [comment]
Super-useful! Am I seeing correctly that plus icon for collections is different to the plus icon for labels? Am I correct that they should be the same? :)

File: kahuna/public/js/edits/image-editor.js (hunk 1 above)
Comment (tsop14) [inline kahuna/public/js/edits/image-editor.js]: For funsies? To make it easier to read? Er...

**Comment by tsop14** [comment]
@paperboyo 
screen shot 2016-02-24 at 11 33 21

**Comment by paperboyo** [comment]
Thanks, @tsop14. Have to buy meself glasses... :eyeglasses:

**Comment by tsop14** [comment]
Haha, they were different! Now they are the same though... and that's all that matters.

File: kahuna/public/js/edits/image-editor.js (hunk 1 above)
Comment (jamesgorrie) [inline kahuna/public/js/edits/image-editor.js]: I would argue moving towards this way as it's stops having funny-little-strings-that-can-change-at-any-time lying all over the place.

Strings are for sentences.

Comment (jamesgorrie) [inline kahuna/public/js/edits/image-editor.js]: And JSON.

File: kahuna/public/js/edits/image-editor.js [hunk 2]
```diff
@@ -115,6 +121,74 @@ imageEditor.controller('ImageEditorCtrl', [
...
+    ctrl.collectionError = false;
+
+    ctrl.selectionMode = true;
+
+    ctrl.openCollectionTree = () => {
+        ctrl.addCollection = true;
+
+        if (!ctrl.collections) {
+            collections.getCollections().then(collections => {
+                ctrl.collections = collections.data.children;
```
Comment (jamesgorrie) [inline kahuna/public/js/edits/image-editor.js]: You could memoize this in the API, and then not have to worry about performance here.

File: kahuna/public/js/edits/image-editor.js (hunk 1 above)
Comment (theefer) [inline kahuna/public/js/edits/image-editor.js]: > Strings are for sentences.

:-)

File: kahuna/public/js/edits/image-editor.js (hunk 2 above)
Comment (tsop14) [inline kahuna/public/js/edits/image-editor.js]: Sounds good - can you vary up how often the result gets reset? What happens if you're trying to add to collection someone else recently created?

File: kahuna/public/js/edits/image-editor.js [hunk 3]
```diff
@@ -21,20 +25,46 @@ imageEditor.controller('ImageEditorCtrl', [
...
+    ctrl.categories;
     ctrl.error = false;
+    ctrl.image;
+    ctrl.saved = false;
+    ctrl.saving = false;
+    ctrl.showUsageRights;
+    ctrl.status = ctrl.image.data.valid ? 'ready' : 'invalid';
+    ctrl.usageRights = imageService(ctrl.image).usageRights;
+    ctrl.usageRightsCategory;
+
```
Comment (tsop14) [inline kahuna/public/js/edits/image-editor.js]: Read you should have all the bindables at the top, so gave it a go.

**Comment by tsop14** [comment]
Colours 💜💛💚💙

screen shot 2016-02-25 at 12 34 00

File: kahuna/public/js/edits/image-editor.js (hunk 2 above)
Comment (tsop14) [inline kahuna/public/js/edits/image-editor.js]: > You could memoize this in the API, and then not have to worry about performance here.

This is already being done 😁

**Comment by NickPapacostas** [comment]
:+1:

**Pull request merged**